pom.xml

```diff
@@ -48,10 +48,12 @@
     <log4j.version>1.2.17</log4j.version>
     <slf4j-log4j12.version>1.7.6</slf4j-log4j12.version>
     <!--
-      We use an old version of Guava to be compatible with Spark 1.1.
-      Check with the spark-cassandra-connector team before upgrading this.
+      Warning: since JAVA-721, Guava version MUST be >= 16.0.1
+      https://code.google.com/p/guava-libraries/issues/detail?id=1635
+      https://code.google.com/p/guava-libraries/issues/detail?id=1637
+      https://bugs.openjdk.java.net/browse/JDK-8031984
     -->
-    <guava.version>14.0.1</guava.version>
+    <guava.version>16.0.1</guava.version>
```

Comment:
:+1: my only worry is that this can bring compatibility issues for users, but I think this is acceptable given this is a major release, and that guava is *mostly* backwards compatible (and in the places its not it should be easy to work around).